File: livebox/src/main/java/com/creations/livebox/util/OkioUtils.java
package com.creations.livebox.util;

import org.apache.commons.io.input.ReaderInputStream;

import java.io.InputStream;
import java.io.OutputStream;
import java.io.Reader;
import java.io.StringReader;
import java.nio.charset.Charset;

import okio.BufferedSource;
import okio.Okio;

/**
 * @author Sérgio Serra on 25/08/2018.
 * Criations
 * sergioserra99@gmail.com
 */
public class OkioUtils {

    public static ReaderInputStream readerInputStreamUtf8(String source) {
        return new ReaderInputStream(new StringReader(source), Charset.forName("UTF-8"));
    }

    public static BufferedSource bufferedSource(InputStream is) {
        return Okio.buffer(Okio.source(is));
    }

    public static void copy(InputStream is, OutputStream os) {

    }
}

File: livebox-common/src/main/java/com/creations/livebox_common/util/OkioUtils.kt
package com.creations.livebox_common.util

import okio.BufferedSource
import okio.Okio
import java.io.InputStream
import java.io.StringReader
import java.nio.charset.Charset

/**
 * Represents the end-of-file (or stream).
 */
const val EOF = -1

fun readerInputStreamUtf8(source: String): ReaderInputStream =
        ReaderInputStream(StringReader(source), Charset.forName("UTF-8"))

fun bufferedSource(`is`: InputStream): BufferedSource = Okio.buffer(Okio.source(`is`))
